Doc 1148/multiple accounts (#413)
swan-io/docs: docs/preview/multiple-accounts/guide-multiple-accounts-integration.mdx

---
title: Create multiple accounts
---

> This guide explains how to open additional accounts for one account holder. 

Reviewer: ```suggestion
This guide explains how to open additional accounts linked to an account holder. 
```
Does this sound better?

## Guide {#guide}

1. Call the `openAccount` mutation.
2. Include the `accountHolderId`  (line 4).
3. Add the `language` parameter (optional) to select the [account language](/topics/accounts/#language). If not specified, English (`en`) is used as the default language (line 5).
4. Add a `name` parameter (optional) for the account. If not specified, "My account" is used as the default name (line 6).

> 5. Add the `OpenAccountSuccessPayload` payload (line 9) and any information you'd like to receive about the additional account.

Reviewer: ```suggestion
5. Use the `OpenAccountSuccessPayload` payload (line 9) to specify the information you'd like to receive about the additional account.
```

> ```graphql {2-3} showLineNumbers

Reviewer: ```suggestion
```graphql {2,4} showLineNumbers
```

  openAccount(
    input: {
      accountHolderId: "$ACCOUNT_HOLDER_ID"
      language: en
      name: "Taxes"
    }
  ) {
    ... on OpenAccountSuccessPayload {
      __typename
      account {
        BIC
        IBAN
        country
        currency
        id
        language
        name
        paymentLevel
        paymentAccountType
        statusInfo {
          status
        }
      }
    }
    ... on InternalErrorRejection {
      __typename
      message
    }
    ... on ForbiddenRejection {
      __typename
      message
    }
    ... on AccountHolderNotFoundRejection {
      __typename
      message
    }
    ... on AccountHolderTypeNotEligibleRejection {
      __typename
      message
    }
    ... on AccountHolderStatusNotEligibleRejection {
      __typename
      message
    }
    ... on AccountHolderVerificationStatusNotEligibleRejection {
      __typename
      message
    }
    ... on AccountHolderAccountsTypeNotEligibleRejection {
      __typename
      message
    }
    ... on AccountHolderAccountsStatusNotEligibleRejection {
      __typename
      message
    }
    ... on AccountMembershipNotEligibleRejection {
      __typename
      message

>     }

Reviewer: Are all these rejections specific to the mutation?
I have doubts on:
- AccountHolderNotFoundRejection
- AccountHolderTypeNotEligibleRejection
- AccountHolderStatusNotEligibleRejection
- AccountHolderVerificationStatusNotEligibleRejection
- AccountHolderAccountsTypeNotEligibleRejection
- AccountHolderAccountsStatusNotEligibleRejection
- AccountMembershipNotEligibleRejection
- InternalErrorRejection
- ForbiddenRejection

  }
}
```

## Payload {#payload}

```json {} showLineNumbers
{
  "data": {
    "openAccount": {
      "__typename": "ForbiddenRejection",

>       "message": "Opening an additional account will be available soon"

Reviewer: We should have a success example here. 
We can also provide rejection examples, but in this case don't we have more helpful rejections to showcase? I don't see this one ever happening when the feature is live
